heuristic/repair_operators/_yth_best_insert.py

```diff
@@ -0,0 +1,41 @@
+import heapq
+from numpy.random import RandomState
+
+from heuristic.classes import Route, Solution
+from heuristic.functions import create_single_customer_route
+
+
+def _yth_best_insert(min_offset: int, current: Solution,
+                     rnd_state: RandomState) -> Solution:
+    """
+    Sequentially inserts a random permutation of the unassigned customers
+    into their y-th best, feasible route at a locally optimal leg of the tour.
+    """
+    rnd_state.shuffle(current.unassigned)
+
+    while len(current.unassigned) != 0:
+        customer = current.unassigned.pop()
+        feasible_routes = []
+
+        for route in current.routes:
+            insert_idx, cost = route.opt_insert(customer)
+
+            if route.can_insert(customer, insert_idx):
+                heapq.heappush(feasible_routes, (cost, insert_idx, route))
+
+        if len(feasible_routes) != 0:
+            offset = min(min_offset, len(feasible_routes))
+            nsmallest = heapq.nsmallest(offset, feasible_routes)
+
+            cost, insert_idx, route = nsmallest[rnd_state.choice(offset)]
```

Comment:
Why? Does `rnd_state.choice(nsmallest)` not work?

Reply:
No, I cant get choice to work for things that are not simple lists or ints.  (This is a list of tuples). ``rnd_state.choice(nsmallest)`` gives the following error:
`` 
line 29, in _yth_best_insert
    cost, insert_idx, route = rnd_state.choice(nsmallest)
  File "mtrand.pyx", line 898, in numpy.random.mtrand.RandomState.choice
ValueError: a must be 1-dimensional
``
Im unsure how to get this to work for a list of tuples, or for a random value from a set.